Source/Dafny/Compilers/Compiler-go.cs

```diff
@@ -2690,13 +2693,17 @@ private class SpecialNativeType : UserDefinedType {
       wr.Write("_dafny.SeqCreate(");
       TrExpr(expr.N, wr, inLetExprBody);
       wr.Write(", ");
-      var fromType = (UserDefinedType)expr.Initializer.Type.NormalizeExpand();
+      var rawFromType = expr.Initializer.Type.NormalizeExpand();
+      var fromType = (UserDefinedType)rawFromType;
```

Comment:
```suggestion
      var fromType = (ArrowType)expr.Initializer.Type.NormalizeExpand();
```

This is simpler and is how it's done for JavaScript below.